examples/ibe.rs

```diff
@@ -0,0 +1,61 @@
+extern crate bn;
+extern crate rand;
+extern crate sodiumoxide;
+
+mod sighash;
+
+use bn::*;
+use sodiumoxide::crypto::stream::chacha20;
+use sodiumoxide::crypto::hash::sha256;
+use std::str;
+
+fn main() {
+  ibe();
+}
+
+fn ibe() {
+    let rng       = &mut rand::thread_rng();
+    let master_sk = Scalar::random(rng);
+    // do we need another generator than G1::one() here?
+    // we use G1 since Ppub is used in the first arg of pairing
+    let master_pk = G1::one() * &master_sk;
+
+    let id = b"test";
+
+    let derived = G2::random(&mut sighash::SignatureHash::from(&id[..]));
+    println!("derived: {:?}", derived);
+
+    let id_sk = &derived * &master_sk;
+
+    //encrypting with BasicIdent
+    let r = Scalar::random(rng);
+    let g_id = pairing(&master_pk, &derived) ^ &r;
+    println!("g_id: {:?}", g_id);
+    let badly_serialized = format!("{:?}", g_id);
```

Comment:
I suggest we break out a `point_to_symmetric_key` function that does sha256(serialized point)